File: public/js/games.js
$(document).ready(function () {
	$('.submit_options').click(function(){
		send_ajax_sport();
	});

	$('.sunday').click(function(){
		send_ajax_day('Sunday');
	})
	$('.monday').click(function(){
		send_ajax_day('Monday');
	})
	$('.tuesday').click(function(){
		send_ajax_day('Tuesday');
	})
	$('.wednesday').click(function(){
		send_ajax_day('Wednesday');
	})
	$('.thursday').click(function(){
		send_ajax_day('Thursday');
	})
	$('.friday').click(function(){
		send_ajax_day('Friday');
	})
	$('.saturday').click(function(){
		send_ajax_day('Saturday');
	})

	$('.reset').click(function(){
		send_ajax_all();
	})

	$( ".register_game" ).click(function() {
  		$( ".create_game_form" ).toggle( "slow", function() {
 	 	});
  	});
});

function send_ajax_all(){
	var ajaxRequest = $.ajax({
		url: '/games/display_all',
		type: 'GET'
	});
	ajaxRequest.done(function(response){
		select_games(response);
	})
}

function send_ajax_day(day){
	var ajaxRequest = $.ajax({
		url: '/games/display_day',
		type: 'GET',
		data: {day: day}
	});
	ajaxRequest.done(function(response){
		select_games(response);
	})
}

function send_ajax_sport(){	
	var sport = $('select').val();
	var ajaxRequest = $.ajax({
		url: '/games/display_sport',
		type: 'GET',
		data: {sport: sport}
	});
	ajaxRequest.done(function(response){
		select_games(response);
	})
}

function select_games(response){
	var new_all_games = $('.all_games').clone();
	$('.all_games').empty();
	console.log(response);
	var length = response.length;
	for (var i = 0; i < length; i ++){
		var sport = response[i].sport;
		var place = response[i].location;
		var game_id = response[i].id;
		var weekday = response[i].weekday;
		var month_day = response[i].month_day;
		var time = response[i].time;
		var date = weekday + ' ' + month_day + ' ' + time 
		var game_details = "<a href='/event/" + game_id + "'><div class='single_game'>" + sport + '<br>' + place + '<br>' + date + '</div>' + '</a>';
		$('.all_games').append(game_details);
	}
}

File: app/controllers/games.rb
get '/games' do 
  @sport = Game.order(:date)
  erb :games
end

get '/games/display_sport' do 
	@sport = Game.where(:sport => params[:sport]).order(:date)
	content_type :json
	@sport.to_json
end

get '/games/display_day' do 
	@sport = Game.where(:weekday => params[:day]).order(:date)
		
	content_type :json
	@sport.to_json
end

get '/games/display_all' do 
	@sport = Game.order(:date)
		
	content_type :json
	@sport.to_json
end

post '/create' do
  p params[:time]
  p params[:date].empty?
  if !params[:date].empty?
	  weekday = Game.findDayOfWeek(params[:date])
	  month_day = Game.findMonthDay(params[:date])
  end
  @new_game = Game.create(sport: params[:sport], location: params[:location], date: params[:date], weekday: weekday, month_day: month_day, time: params[:time])
  if @new_game.errors.full_messages.nil?
    redirect '/games'
  else
  	@games = Game.all
    erb :index
  end
end
